### ISSUE TITLE
Adding jruby-truffle.jar breaks on Windows

### ISSUE THREAD

**Issue by jkutner** [jkutner/mjruby#9]
The org/jruby/Main class can't be found when truffle is on the boot classpath.

**Comment by jkutner** [comment]
See: https://github.com/jkutner/mjruby/blob/master/mrblib/jruby_support.rb#L73

**Comment by chrisseaton** [comment]
Truffle doesn't work on Windows due to some JNR calls we do - but that doesn't explain what would cause this problem.

Is there more error output? It might be more complex than simply 'not found'.

**Comment by jkutner** [comment]
No matter how I twist this around, I still get NULL returned from the [JNI call to FindClass](https://github.com/jkutner/mruby-jvm/blob/afbe4f498899cac95251d3614218995870d4511f/src/java_support.c#L103). Effectively:

``` c
jclass main_class = (*env)->FindClass(env, "org/jruby/Main");
```

@chrisseaton any ideas here?

**Comment by chrisseaton** [comment]
What I'm still not sure about is does removing Truffle from the class path completely solve the problem for you?

**Comment by jkutner** [comment]
@chrisseaton yes. i just toggle this line: https://github.com/jkutner/mjruby/blob/master/mrblib/jruby_support.rb#L73

only happens on Windows. I've tried reordering the jars on the classpath. no change. I'm mainly searching for a way to get more info, not necessarily a solution.

I could also exclude the truffle JAR from the boot classpath if I detected that it's windows. Would that be ok?

**Comment by chrisseaton** [comment]
Can I just suggest you keep Truffle off the bootclasspath for now then and we'll treat Truffle as an unsupported feature in the new launcher for now?

I'd have to reproduce to do anything more helpful, which would mean setting up a Windows dev environment that I don't immediately have. I will add it to my todo list but can't promise anything.

It is certainly a strange bug. Truffle may seem like magic but jruby-truffle.jar doesn't include anything weird like our JIT compiler - it's just Java code with the exception as the same JNR stuff that normal JRuby uses.

**Comment by jkutner** [comment]
@chrisseaton yea, i think leaving it off the path is perfect for. i just wasn't sure if that was ok. we can deal with this later.
